src/main/java/ch/njol/skript/events/EvtMove.java

```diff
@@ -0,0 +1,104 @@
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.log.ErrorQuality;
+import ch.njol.util.coll.CollectionUtils;
+import io.papermc.paper.event.entity.EntityMoveEvent;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerMoveEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+public class EvtMove extends SkriptEvent {
+
+	private static final boolean HAS_ENTITY_MOVE = Skript.classExists("io.papermc.paper.event.entity.EntityMoveEvent");
+
+	// There was a point the event existed but this method did not
+	private static final boolean HAS_POS_METHOD = HAS_ENTITY_MOVE &&
+		Skript.methodExists(EntityMoveEvent.class, "hasChangedPosition", boolean.class);
+
+	static {
+		Class<? extends Event>[] events;
+		if (HAS_ENTITY_MOVE)
+			events = CollectionUtils.array(PlayerMoveEvent.class, EntityMoveEvent.class);
+		else
+			events = CollectionUtils.array(PlayerMoveEvent.class);
+
+		Skript.registerEvent("Move", EvtMove.class, events,
+			"(0¦player|1¦%entitydatas%) move")
+			.description("Called when a player or entity moves.",
+				"NOTE: Entity move event will only be called when the entity moves position, not orientation (ie: looking around).",
+				"NOTE: These events can be performance heavy as they are called quite often.",
+				"If you use these events, and later remove them, a server restart is recommended to clear registered events from Skript.")
+			.examples("on player move:",
+				"\tif player does not have permission \"player.can.move\":",
+				"\t\tcancel event",
+				"on skeleton move:",
+				"\tif event-entity is not in world \"world\":",
+				"\t\tkill event-entity")
+			.requiredPlugins("Paper 1.16.5+ (for entity move)")
+			.since("INSERT VERSION");
+	}
+
+	@Nullable
+	private EntityData<?>[] types = null;
+
+	@Override
+	public boolean init(Literal<?>[] args, int matchedPattern, SkriptParser.ParseResult parseResult) {
+		if (parseResult.mark == 1) {
+			if (HAS_ENTITY_MOVE)
+				types = ((Literal<EntityData<?>>) args[0]).getAll();
+			else {
+				Skript.error("Entity move event requires Paper 1.16.5+", ErrorQuality.SEMANTIC_ERROR);
+				return false;
+			}
+		}
+		return true;
+	}
+
+	@Override
+	public boolean check(Event event) {
+		if (types == null && event instanceof PlayerMoveEvent) {
+			return true;
+		} else if (types != null && HAS_ENTITY_MOVE && event instanceof EntityMoveEvent) {
+			EntityMoveEvent entityEvent = (EntityMoveEvent) event;
+			LivingEntity entity = entityEvent.getEntity();
+			for (EntityData<?> type : types) {
+				if (type.isInstance(entity)) {
+					if (!HAS_POS_METHOD || entityEvent.hasChangedPosition()) {
```

Comment:
I think the check should be  done regardless, after all if I did understand correctly that new method is just a helper method for:
```java
Location from = event.getFrom();
Location to = event.getTo();

if (from.getX() != to.getX()
  || from.getY() != to.getY()
  || from.getZ() != to.getZ()
  || !from.getWorld().equals(to.getWorld()))
```

And this is out of the scope of this PR to an extent but if you want, new conditions could be added for those checks and fallback to the helper methods if they exist.